code/lib/create-storybook/src/bin/index.ts

```diff
@@ -1,28 +1,60 @@
-import { versions } from 'storybook/internal/common';
+/* eslint-disable no-underscore-dangle */
 import { addToGlobalContext } from 'storybook/internal/telemetry';
 
 import { program } from 'commander';
-import { findPackageSync } from 'fd-package-json';
-import invariant from 'tiny-invariant';
 
+import { version } from '../../package.json';
 import type { CommandOptions } from '../generators/types';
 import { initiate } from '../initiate';
+import { modernInputs } from './modernInputs';
 
-addToGlobalContext('cliVersion', versions.storybook);
+const IS_NON_CI = process.env.CI !== 'true';
+const IS_NON_STORYBOOK_SANDBOX = process.env.IN_STORYBOOK_SANDBOX !== 'true';
 
-const pkg = findPackageSync(__dirname);
-invariant(pkg, 'Failed to find the closest package.json file.');
+addToGlobalContext('cliVersion', version);
 
-program
-  .name('Initialize Storybook into your project.')
+const createStorybookProgram = program.name('Initialize Storybook into your project.');
+
+const modernProgram = Object.entries(modernInputs.shape).reduce((acc, [key, schema]) => {
+  const { innerType, defaultValue, description } = schema._def;
+  let t = innerType;
+
+  while (t._def.innerType) {
+    // @ts-expect-error (Object.entries loses type information)
+    t = t._def.innerType;
+  }
+
+  const { typeName } = t._def;
+  const value = defaultValue();
+
+  // to dash-case
+  const flag = key
+    .split(/(?=[A-Z])/)
+    .join('-')
+    .toLowerCase();
+
+  if (typeName.match('String')) {
+    acc.option(`--${flag} <option>`, description, value);
+  } else if (typeName.match('Enum')) {
+    acc.option(`--${flag} <option>`, description, value);
+  } else if (typeName.match('Boolean')) {
+    acc.option(`--${flag}`, description, value);
+    if (value === true) {
+      acc.option(`--no-${flag}`, `inverted --${flag}`);
+    }
+  }
```

Comment:
Can't we extend `modernInputs` to include the legacy formats, so we don't have to map them here?